crashtracker-ffi/src/collector_windows/api.rs

```diff
@@ -0,0 +1,836 @@
+    // HKEY_CURRENT_USER. If not, create it in HKEY_CURRENT_USER.
+
+    // Convert value name to null-terminated wide string
+    let mut name_wide: Vec<u16> = path.encode_utf16().collect();
+    name_wide.push(0); // Add null terminator
+    let name_pcwstr = PCWSTR::from_raw(name_wide.as_ptr());
+
+    // Subkey path as wide string constant
+    let subkey =
+        w!("SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\RuntimeExceptionHelperModules");
+
+    // Check both HKEY_LOCAL_MACHINE and HKEY_CURRENT_USER
+    for root in [HKEY_LOCAL_MACHINE, HKEY_CURRENT_USER] {
+        let mut hkey = HKEY::default();
+
+        // Try to open the registry key
+        let open_result = unsafe { RegOpenKeyExW(root, subkey, None, KEY_READ, &mut hkey) };
+
+        if open_result == ERROR_SUCCESS {
+            // Check if the value exists
+            let query_result = RegQueryValueExW(hkey, name_pcwstr, None, None, None, None);
+
+            let _ = RegCloseKey(hkey);
+
+            if query_result == ERROR_SUCCESS {
+                // Value exists in either hive, exit successfully
+                return Ok(());
+            }
+        }
+    }
+
+    // Value doesn't exist in either hive, create in HKEY_CURRENT_USER
+    let mut hkey = HKEY::default();
+
+    // Create or open the key with write access
+    let create_result = RegCreateKeyExW(
+        HKEY_CURRENT_USER,
+        subkey,
+        None,
+        None,
+        REG_OPTION_NON_VOLATILE,
+        KEY_WRITE,
+        None,
+        &mut hkey,
+        None,
+    );
+
+    anyhow::ensure!(
+        create_result == ERROR_SUCCESS,
+        "Failed to create registry key {path}"
+    );
+
+    // Create the DWORD value (0)
+    let dword_value: u32 = 0;
+    let dword_bytes = dword_value.to_ne_bytes();
+    let set_value_result = RegSetValueExW(hkey, name_pcwstr, None, REG_DWORD, Some(&dword_bytes));
+
+    anyhow::ensure!(
+        set_value_result == ERROR_SUCCESS,
+        "Failed to set registry value"
+    );
+
+    let _ = RegCloseKey(hkey);
+
+    Ok(())
+}
+
+unsafe fn output_debug_string(message: &str) {
+    let mut wstr: Vec<u16> = message.encode_utf16().collect();
+    wstr.push(0); // Ensure null termination
+    OutputDebugStringW(PCWSTR::from_raw(wstr.as_ptr()));
+}
+
+fn set_error_context(message: &str) {
+    let bytes = message.as_bytes();
+    let boxed_slice = bytes.to_vec().into_boxed_slice();
+    let static_slice = Box::leak(boxed_slice);
+
+    unsafe {
+        WERCONTEXT.ptr = static_slice.as_ptr();
+        WERCONTEXT.len = static_slice.len();
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ErrorContext {
+    endpoint: Option<Endpoint>,
+    metadata: datadog_crashtracker::Metadata,
+}
+
+static mut WERCONTEXT: WerContext = WerContext {
+    prefix: WER_CONTEXT_PREFIX,
+    ptr: std::ptr::null(),
+    len: 0,
+    suffix: WER_CONTEXT_SUFFIX,
+};
+
+pub const WER_CONTEXT_PREFIX: u64 = 0xBABA_BABA_BABA_BABA;
```

Comment:
Is there meaning to this bitpattern? Or can it be any, and this is just easy to see in a debugger?

Reply:
No meaning, I just needed a non-zero pattern. In the future it could be repurposed for versioning if needed. Gonna add a comment.